dace/codegen/targets/cpu.py

```diff
@@ -1109,6 +1109,9 @@ def memlet_definition(self,
         elif var_type in [DefinedType.Stream, DefinedType.StreamArray]:
             if not memlet.dynamic and memlet.num_accesses == 1:
                 if not output:
+                    index = cpp.cpp_offset_expr(desc, memlet.subset)
+                    if isinstance(desc, data.Stream) and desc.is_stream_array():
+                        expr = f"{memlet.data}[{index}]"
```

Comment:
You can move the computation of index into the branch, since it's not needed otherwise

Reply:
True!